crates/rpc/eth/logs.rs

```diff
@@ -0,0 +1,180 @@
+use crate::{
+    types::{block_identifier::BlockIdentifier, receipt::RpcLog},
+    RpcErr, RpcHandler,
+};
+use ethereum_rust_core::{H160, H256};
+use ethereum_rust_storage::Store;
+use serde::Deserialize;
+use serde_json::Value;
+use std::collections::HashSet;
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+pub enum AddressFilter {
+    Single(H160),
+    Many(Vec<H160>),
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(untagged)]
+pub enum TopicFilter {
+    Topic(Option<H256>),
+    Topics(Vec<Option<H256>>),
+}
+
+#[derive(Debug)]
+pub struct LogsRequest {
+    /// The oldest block from which to start
+    /// retrieving logs.
+    /// Will default to `latest` if not provided.
+    pub from_block: BlockIdentifier,
+    /// Up to which block to stop retrieving logs.
+    /// Will default to `latest` if not provided.
+    pub to_block: BlockIdentifier,
+    /// The addresses from where the logs origin from.
+    pub address_filters: Option<AddressFilter>,
+    /// Which topics to filter.
+    pub topics: Vec<TopicFilter>,
+}
+impl RpcHandler for LogsRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<LogsRequest, RpcErr> {
+        match params.as_deref() {
+            Some([param]) => {
+                let param = param.as_object().ok_or(RpcErr::BadParams)?;
+                let from_block = {
+                    if let Some(param) = param.get("fromBlock") {
+                        BlockIdentifier::parse(param.clone(), 0)?
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let to_block = {
+                    if let Some(param) = param.get("toBlock") {
+                        BlockIdentifier::parse(param.clone(), 1)?
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let address_filter = param.get("address").and_then(|address| match address {
+                    Value::String(_) | Value::Array(_) => {
+                        Some(serde_json::from_value::<AddressFilter>(address.clone()).unwrap())
+                    }
+                    _ => None,
+                });
+                let topics = param.get("topics").map(|topics| {
+                    serde_json::from_value::<Option<Vec<TopicFilter>>>(topics.clone()).unwrap()
```

Comment:
unwrap or return bad params?

Reply:
[I've handled the unwrap here](https://github.com/lambdaclass/ethereum_rust/pull/360/commits/07e330379c5f21d3b533ae5f63c6f62ed1eefccd#diff-6c9706e673071c6e5c00ad9c3289d020db454f364f2e3da5c392bdefb5393dc9R51), wdyt?

Reply:
yes but for the topics you can do `.map_err(|_| RpcErr::BadParams))` instead of panicking with the unwrap

Reply:
You're correct, I got caught up with the other comment, [something like this avoids the panic](https://github.com/lambdaclass/ethereum_rust/pull/360/commits/04adf2f513b025ca36df6fa31049a08b9b2f7d6d).

I tried the map_err solution for a bit, but that opened a can of worms where RpcErr was missing deserialize and went with a match instead. A bit more of lines, but it works.